azkaban-execserver/src/main/java/azkaban/execapp/StatisticsServlet.java

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.execapp;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.log4j.Logger;
+
+import azkaban.executor.Statistics;
+import azkaban.utils.JSONUtils;
+
+public class StatisticsServlet extends HttpServlet  {
+  private static final long serialVersionUID = 1L;
+  private static final int  cacheTimeInMilliseconds = 1000;
+  private static final Logger logger = Logger.getLogger(StatisticsServlet.class);
+
+  protected static Date lastRefreshedTime = null;
+  protected static Statistics cachedstats = null;
+
+  /**
+   * Handle all get request to Statistics Servlet {@inheritDoc}
+   *
+   * @see javax.servlet.http.HttpServlet#doGet(javax.servlet.http.HttpServletRequest,
+   *      javax.servlet.http.HttpServletResponse)
+   */
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+      throws ServletException, IOException {
+
+    if (null == lastRefreshedTime ||
+        new Date().getTime() - lastRefreshedTime.getTime() > cacheTimeInMilliseconds){
```

Comment:
Can we use System.currentTimeMillis() to achieve the same goal but w/o creating new object?